util/errors/errors.go

```diff
@@ -93,6 +94,22 @@ func isTransientEtcdErr(err error) bool {
 	return false
 }
 
+func isTransientInterruptionErr(err error) bool {
+	// Handle transient node/resource related errors
+	if strings.Contains(err.Error(), "imminent node shutdown") {
+		return true
+	} else if strings.Contains(err.Error(), "node was low on resource") {
+		return true
+	} else if strings.Contains(err.Error(), "pod deleted") {
```

Comment:
This is a reference to this error, right? https://github.com/argoproj/argo-workflows/blob/e0a0f2856ba6e8a49b1ea28b15d50d39f3106125/workflow/controller/operator.go#L1262

From looking at that method, it seems this error should only happen during reconciliation when a pod is deleted after the `RECENTLY_STARTED_POD_DURATION` period without the controller being informed, which is unusual. Have you been encountering that under normal operation?